include/swift/Runtime/Concurrency.h

```diff
@@ -219,8 +312,20 @@ bool swift_task_tryAddStatusRecord(AsyncTask *task,
 ///s
 /// Returns false if the task has been cancelled.
 SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
-bool swift_task_removeStatusRecord(AsyncTask *task,
-                                   TaskStatusRecord *record);
+bool swift_task_removeStatusRecord(AsyncTask *task, TaskStatusRecord *record);
+
+/// Attach a child task to its parent task and return the newly created
+/// `ChildTaskStatusRecord`.
+///
+/// The record must be removed with by the parent invoking
+/// `swift_task_detachChild` when the child has completed.
+SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
+ChildTaskStatusRecord*
+swift_task_attachChild(AsyncTask *parent, AsyncTask *child);
+
+/// Remove a child task from the parent tracking it.
+SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
+void swift_task_detachChild(AsyncTask *parent, ChildTaskStatusRecord *record);
```

Comment:
Which of these functions actually need to be ABI, vs. potentially being internal to the concurrency library?

Reply:
I don't really know is the honest answer.

We need to call this or something similar when we spawn a child, be it from the SIL work that does the `async let` child task creation etc, or from the `group.add` function.

If it is not meant to be ABI then the intended way to call this functionality would be some private header that only 
"we" import, and we call from those two places? So like `_swift_task_alloc_destroy` that's an example of what a private to the concurrency lib function is?

I can make those private yes I think